### PR TITLE
Global autocomplete: Improve dynamic data support

### PR DESCRIPTION
**Pull request by morgaan**
- Enhance arguments with additional `httpMethod` setting
- Enhance arguments with additional `bodyTemplate` setting
- Add support for the two above new settings
- Delegate "No results" to resultsCallBack rather than generateSuggestions
- Update demo
- Update HISTORY
- Bump to next major version because of breaking change

Forked from the PR #154

### CODE REVIEW

File: toolkits/global/packages/global-autocomplete/demo/main.js [hunk 1]
```diff
@@ -33,13 +37,20 @@ const args = {
...
 	endpoint: 'autocomplete?q=',
-  //staticResultsData: animalsList,	// OPTIONAL: Provide either endpoint or array of data - if both are provided array be used
+	// staticResultsData: animalsList,	// OPTIONAL: Provide either endpoint or array of data - if both are provided array be used
 	timeout: 2000,		// OPTIONAL: Set a timeout for the fetch request, onError will be called if fetch request timeouts, default is 2000
 	minChars: 1,			// OPTIONAL: Minimum characters to be typed before request is sent, default is 0
 	inputDelay: 300,	// OPTIONAL: Delay between keypress and request being sent, default is 300
 	headers: {
 		Accept: 'application/json; version=2'
 	},
+	// httpMethod: 'POST', // OPTIONAL: Set which HTTP method must be used to emit the fetch request to the suggestions endpoint, default is 'GET'
```
Comment (jon-stevens) [inline toolkits/global/packages/global-autocomplete/demo/main.js]: I think 
```suggestion
	// httpMethod: 'POST', // OPTIONAL: Defaults to 'GET'
```
is probably OK?

Comment (morgaan) [reply toolkits/global/packages/global-autocomplete/demo/main.js]: Yes

File: toolkits/global/packages/global-autocomplete/demo/main.js [hunk 2]
```diff
@@ -9,6 +9,10 @@ const showResults = results => {
 	const resultsContainer = document.createElement('div');
 	resultsContainer.className = 'c-results-container';
 
+	// Assuming results is an array
+	if(results.length === 0) {
+		results.push['No results'];
```
Comment (jon-stevens) [inline toolkits/global/packages/global-autocomplete/demo/main.js]:
```suggestion
		results.push('No results');
```

File: toolkits/global/packages/global-autocomplete/demo/main.js [hunk 3]
```diff
@@ -33,13 +37,20 @@ const args = {
...
-  //staticResultsData: animalsList,	// OPTIONAL: Provide either endpoint or array of data - if both are provided array be used
+	// staticResultsData: animalsList,	// OPTIONAL: Provide either endpoint or array of data - if both are provided array be used
 	timeout: 2000,		// OPTIONAL: Set a timeout for the fetch request, onError will be called if fetch request timeouts, default is 2000
 	minChars: 1,			// OPTIONAL: Minimum characters to be typed before request is sent, default is 0
 	inputDelay: 300,	// OPTIONAL: Delay between keypress and request being sent, default is 300
 	headers: {
 		Accept: 'application/json; version=2'
 	},
+	// httpMethod: 'POST', // OPTIONAL: Defaults to 'GET'
+	// bodyTemplate: term => { // OPTIONAL: Object template to build the body to be passed into the fetch request to the suggestions endpoint
```
Comment (jon-stevens) [inline toolkits/global/packages/global-autocomplete/demo/main.js]:
```suggestion
	// bodyTemplate: term => { // OPTIONAL: template for the body data
```

**Pull request merged**